File: project-euler/js/007.js
const { parse } = require('path');
const readline = require('readline')

const rl = readline.createInterface({input: process.stdin, output: process.stdout});
rl.prompt();

let input = '' 

rl.on('line', (inputLine) => {
    input+=(inputLine+'\n');
})

rl.on('close', () => {
    main(input);
    process.exit(0);
})

function fnIsPrime(n) {
    if (n === 0 || n === 1 || (n % 2 === 0 && n !== 2)) { 
        return false 
    }
    for (let i = 3; i < Math.ceil(Math.sqrt(n)) + 1; i++) {
        if (n % i === 0) { return false }
    }
    return true
}

function nthPrime(n) {
    let i = 0
    while (n !== 0) {
        i++
        if (isPrime(i)) {
            n--
        }   
    }
    return i
}

// The code above works for 4/5 cases... I dont see anything majorly different from that below...
function nthPrimeFactor(n) {
    let primes = [2]
    for (i = 3; primes.length < n; i += 2) {
        let isPrime = true
        for (let j = 3; j <= Math.ceil(Math.sqrt(i))+1; j += 2) {
            if (i%j === 0 && i !== j) {
                isPrime = false
                break
            }
        }
        if (isPrime) {
            primes.push(i)
        }
    }
    return primes[primes.length-1]
}

function main(input) {

    const inputArray = input
        .split('\n')
        .filter(elem => elem !== '')
        .map(curr => parseInt(curr) )
    const numInputs = inputArray[0];

    for(var i = 1; i < numInputs+1; i++) {
        console.log(nthPrimeFactor(inputArray[i]))
    }

}

File: project-euler/python/007.py
#!/bin/python3
import sys

#This works for 3/5 cases (not sure why it doesnt work for test cases 3 and 5)
def nth_prime(n): 
    #this is the same implementation as that in JS with the is_prime check with in the loop
    primes = [2]
    i = 3
    while len(primes) < n:
        is_prime = True
        j = 3
        while j <= int(i**0.5)+1:
            if i % j == 0 and i != j:
                is_prime = False
                break
            j += 2
        if is_prime:
            primes.append(i)        
        i += 2
    return primes[-1]

num_inputs = int(input().strip())
for _ in range(num_inputs):
    n = int(input().strip())
    print(nth_prime(n))